tests/e2e/scripts/latest_commit.sh

```diff
@@ -1,7 +1,25 @@
 #!/bin/bash
-# Grabs the last 5 commit SHA's from the given branch, then purges any commits that do not have a passing CI build
+
+branch=$1
+output_file=$2
+# Grabs the last 10 commit SHA's from the given branch, then purges any commits that do not have a passing CI build
 iterations=0
-response=$(curl -s -H 'Accept: application/vnd.github.v3+json' "https://api.github.com/repos/rancher/rke2/commits?per_page=5&sha=$1")
+
+# The VMs take time on startup to hit aws, wait loop until we can
+while ! curl -s --fail https://rke2-ci-builds.s3.amazonaws.com > /dev/null; do
```

Comment:
nit: save downloading the WHOLE object list from S3, we just want to see if we can hit it or not
```suggestion
while ! curl -s --fail https://rke2-ci-builds.s3.amazonaws.com?max-keys=0 > /dev/null; do
```

Reply:
Nice catch